### ISSUE TITLE
Improve tests for source info - suspected bug

### ISSUE THREAD

**Issue by craigfowler** [csf-dev/ZPT-Sharp#141]
I have a feeling that the document source info is being lost or miss-set for imported documents; such as when METAL has combined documents together.

I recall seeing an error message which had the wrong source info attached. This could be caught with unit tests.

**Comment by craigfowler** [comment]
The error I saw was not from source annotation; it was from the evaluation of a TALES expression. The source info was reported in an exception message.

**Comment by craigfowler** [comment]
Perhaps the right thing to do is to perform some kind of 'source info burn-in', so that the info is extracted from the source doc when it is first imported and stored with the element. That way, it will be consistent across all usages.

I think this is a similar problem to the one I saw with source annotation. I fixed that by doing it at as different stage in the document's lifetime.

**Comment by craigfowler** [comment]
The above might not be possible as the framework is currently structured. The elements are not persistent so we cannot burn the source info into them.

Getting the source Info again, we know cones with problems (I think that's where this bug arises).

Perhaps the best thing to do is just remove the usage of the source info from the errors. That way we don't do anything particularly wrong, but our errors are less useful.

**Comment by craigfowler** [comment]
Deferred, because I can't find the place where it had previously gone wrong.

**Comment by craigfowler** [comment]
Deferred again, since after #156 I may well have fixed this indirectly.  This issue should now be reviewed and only re-added to a sprint/milestone if it is reproducible after the date of this comment.

**Comment by craigfowler** [comment]
I'm going to close this for now - I can re-open it if I discover a real problem. However, since this was first raised, all of the code in this area has been rewritten.
